### ISSUE TITLE
Validation falls into infinite loop with lodash 4.17.0

### ISSUE THREAD

**Issue by gvidon** [davidkpiano/react-redux-form#535]
I have a field on a form which value is object with nested properties:
```json
{
  "_id": "abc123",
  "first_name": "John",
  "last_name": "Wayne"
}
```

With lodash 4.17.0 this code in our code base falls into infinite loop, infinitely  dispatching `setValidity` action:

```javascript
// Only failing method shown
class Row extends Component {

	// Update model value, which is fetched asynchronously
	// Call form validation action once data updated
	componentWillReceiveProps(nextProps) {
		if(nextProps.value === this.props.value || nextProps.value === this.changedValue)
			return false;

		this.changedValue = nextProps.value;
		this.props.validate();
		this.setPresentationValue(nextProps.value, nextProps);
	}
}
```

There are **no validators** defined for failing fields.

**Comment by davidkpiano** [comment]
```js
this.props.validate();
```

What does this method look like?

**Comment by gvidon** [comment]
@davidkpiano it is action dispatching. We every custom control with this method, using this:

```javascript
// Form row wrapper, provides all data and logic down to wrapped UI component
import {has, isUndefined} from 'lodash';
import React from 'react';
import {actions, Control} from 'react-redux-form';

const
	commonPropsMap = {
		disabled  : props => Boolean(isUndefined(props.fieldValue.disabled) ? props.disabled : props.fieldValue.disabled),
		focus     : props => props.fieldValue.focus,
		isRequired: props => has(props.validators, 'isRequired'),
		onBlur    : props => () => props.dispatch(actions.blur(props.model)),
		valid     : props => props.fieldValue.valid,
		validate  : props => () => props.dispatch(actions.validate(props.model, props.validators))
	},

	createField = (component, mapProps) => {
		const
			Field = props => <Control component={component} mapProps={{...commonPropsMap, ...mapProps}} {...props}>
				{props.children}
			</Control>;

		Field.propTypes = component.propTypes;
		return Field;
	};

export {createField};
```

And apply like this:

```javascript
import ContactRow from 'generic/Form/ContactRow';
import {controls} from 'react-redux-form';
import {createField} from 'generic/Form/Field';
export default createField(ContactRow, controls.select)
```

**Comment by davidkpiano** [comment]
So the issue here is that you are manually validating the model every time your component receives props, whether the model has changed or not. The standard usage of `<Control>` internally has optimizations on when it validates, so that it avoids validating:
- when the model value hasn't changed
- when the validity hasn't changed

I would recommend leaving validation to the `<Control>` component, or figure out where in your custom code it is creating an infinite loop. I would investigate why the value is infinitely changing, even at times where it should be the same. Perhaps do a `shallowEqual` instead?

So long story short, avoid handling validation in `componentWillReceiveProps` or `componentWillUpdate` or `componentDidUpdate`, etc.

**Comment by gvidon** [comment]
@davidkpiano it looks like after `SET_VALIDITY` is dispatched there is no our custom code participating in the process. Loop starts even without changing anything, but during form init. 

The flow is — data fetched, store updated, `componentWillReceiveProps` called, `SET_VALIDITY` falls into loop.

It started right after lodash updated yesterday. Unable to reproduce with lodash < 4.17.0.

**Comment by davidkpiano** [comment]
Weird... I'll update lodash and see if I can reproduce it.

**Comment by perrydt** [comment]
We just ran into a similar problem and by comparing shrink-wrapped versions of our deployed app were able to track it to lodash 4.17.0 as well.  Our affected forms are using custom validation functions in the validators prop, but are all client-side; for us the symptom was not an infinite loop but very poor form performance -- several seconds to handle each keystroke.

However, the problem seems to be gone after updating to lodash 4.17.1 which was released yesterday -- one of the changelog items mentions fixing an "_.omit performance regression".

**Comment by fxfilmxf** [comment]
I noticed the same problem. Several seconds for each keystroke. I traced it down to `omit` which I thought was pretty odd until I saw the issues that were filed against lodash 4.17.0. There's an issue with `pick` in 4.17.1 so I would suggest staying on 4.15 or waiting until 4.17.2 is release later today.

**Comment by davidkpiano** [comment]
For now, I replaced `lodash/omit` with a simpler internal implementation.
